File: #0075.sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {

        int index = 0;
        int size = nums.size();
        for (int i = 0; i < 3; i++) {
            for (int j = index; j < size; j++) {
                if (nums[j] == i) {
                    swap(nums[j], nums[index]);
                    index++;
                }
            }
        }
    }
};

File: #0075.sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    let left = 0;
    let right = nums.length-1;
    let curr = 0;
    while (curr <= right) { 
        if (nums[curr] == 0) { 
            [nums[curr++], nums[left++]] = [nums[left], nums[curr]];
        } else if (nums[curr] == 2) { 
            [nums[curr], nums[right--]] = [nums[right], nums[curr]];
        } else {
            curr++;
        }
    }
};
